BZ1664914: descheduler documentation lack of the introduction of 'nodeaffinity' strategy (#34906)
openshift/openshift-docs: admin_guide/scheduling/descheduler.adoc

<1> Set this value to `enabled: true` to use this policy. Set to `false` to disable this policy.

[[admin-guide-descheduler-policies-node]]
=== Remove Pods Violating Node Affinity

The `RemovePodsViolatingNodeAffinity` strategy ensures that all pods violating node affinity are removed from nodes. This situation could occur if a node no longer satisfies a pod’s affinity rule. If another node is available that satisfies the affinity rule, then the pod is evicted.

> When the `requiredDuringSchedulingIgnoredDuringExecution` node affinity type is enabled, the `RemovePodsViolatingNodeAffinity` strategy ensures serves as a temporary implementation of `requiredDuringSchedulingRequiredDuringExecution` by evicting pods that no longer respects node affinity.

Reviewer: `ensures serves` -> `serves`

<1> Set this value to `enabled: true` to use this policy. Set to `false` to disable this policy.

[[admin-guide-descheduler-policies-node]]
=== Remove Pods Violating Node Affinity

> The `RemovePodsViolatingNodeAffinity` strategy ensures that all pods violating node affinity are removed from nodes. This situation could occur if a node no longer satisfies a pod’s affinity rule. If another node is available that satisfies the affinity rule, the pod is evicted.
> 
> For example, *podA* is scheduled on *nodeA*, because the node satisfies the `requiredDuringSchedulingIgnoredDuringExecution` node affinity rule at the time of scheduling. If *nodeA* stops satisfying the rule, when the strategy gets executed, if there is another node available that satisfies the node affinity rule, *podA* is evicted from *nodeA* and moved to the other node.

Reviewer 2: There are quite a few independent clauses strung together in these paragraphs. It might improve clarity by using active voice where possible. Not sure if my suggested edit still holds the same meaning, but gave it a shot:

```suggestion
The `RemovePodsViolatingNodeAffinity` strategy ensures that all pods violating node affinity are removed from nodes. This situation might occur if a node no longer satisfies a pod’s affinity rule. If another node is available that satisfies the affinity rule, the pod is evicted.

For example, *podA* is scheduled on *nodeA*, because the node satisfies the `requiredDuringSchedulingIgnoredDuringExecution` node affinity rule at the time of scheduling. If *nodeA* stops satisfying the rule, and if there is another node available that satisfies the node affinity rule, the strategy evicts *podA* from *nodeA* and moves it to the other node.
```

    spec:
        containers:
        - name: descheduler

>           image: registry-proxy.engineering.redhat.com/rh-osbs/openshift3-ose-descheduler:v3.11.487-1.git.d435537

Reviewer: Better an image that's outdated rather than an image that can not be pulled. Still, can you mention to check for the latest before deploying the job?

Author: Sorry. Took you literally in the BZ. I updated to `image: registry.access.redhat.com/openshift3/ose-descheduler`. Is that correct?

Reviewer: `docker inspect registry.access.redhat.com/openshift3/ose-descheduler` says (among other bits):
```
       "Created": "2021-07-01T21:08:39.605372657Z",
        "Config": {
            "Env": [
                ...
                "BUILD_RELEASE=1.git.d435537",
                "BUILD_VERSION=v3.11.465",
```
It might still be valid as the latest image is not yet released for consumption. Let's go with it.

Author: I used `image: registry.access.redhat.com/openshift3/ose-descheduler` in my job file and it picked up that image: `registry.access.redhat.com/openshift3/ose-descheduler:v3.11.465-1.git.d435537`
